src/utils/formatUtils.ts

```diff
@@ -15,18 +24,45 @@ export const getNewDiscourseNodeText = ({
   nodeType: string;
   blockUid?: string;
 }) => {
-  const discourseNodes = getDiscourseNodes().filter(
-    (n) => n.backedBy === "user"
-  );
+  const discourseNodes = getDiscourseNodes();
+  let newText = text;
+  if (!text) {
+    newText = await new Promise<string>((resolve) => {
+      const nodeName =
+        discourseNodes.find((n) => n.type === nodeType)?.text || "Discourse";
+      renderFormDialog({
+        title: `Create ${nodeName} Node`,
+        fields: {
+          textField: {
+            type: "text",
+            label: `Create ${nodeName} Node`,
+          },
+        },
+        onSubmit: (data: Record<string, unknown>) => {
+          resolve(data.textField as string);
+        },
+        onClose: () => {},
```

Comment:
if the user just closes this dialog, this function will just get hung up on this line the promise won't resolve